simplq/src/components/pages/Admin/AdminPage.jsx

```diff
@@ -6,18 +6,44 @@ import { useDispatch, useSelector } from 'react-redux';
 import { selectQueueName } from 'store/selectedQueue';
 import Ribbon from 'components/common/Ribbon';
 import Tour from 'components/common/Tour';
-import { useGetSelectedQueue } from 'store/asyncActions';
+import { useGetSelectedQueue, useGetQueueInfo } from 'store/asyncActions';
+import { selectQueueInfo } from 'store/queueInfo';
 import HeaderSection from './AdminHeaderSection';
 import TokenList from './TokenList';
 import styles from './admin.module.scss';
 import SidePanel from './AdminSidePanel';
 import getToursteps from './getTourSteps';
+import PageNotFound from '../PageNotFound';
 
 const TIMEOUT = 10000;
 let timeoutId;
 
-export default (props) => {
+const isQueueDeleteStatusUnknown = (queueInfo) => !queueInfo || Object.keys(queueInfo).length === 0;
+const isQueueDeleted = (queueInfo) => queueInfo.status === 'DELETED';
+
+const AdminPage = (props) => {
   const queueId = props.match.params.queueId;
+  const queueInfo = useSelector(selectQueueInfo);
+  const dispatch = useDispatch();
+  const getQueueInfo = useCallback(useGetQueueInfo(), []);
+
+  useEffect(() => {
+    dispatch(getQueueInfo({ queueId }));
+  }, [dispatch, queueId, getQueueInfo]);
+
+  if (isQueueDeleteStatusUnknown(queueInfo)) {
+    return <></>;
+  }
```

Comment:
Why did we have to do this? Curious to know.

Can we avoid it?

Reply:
By the time this useSelector hook fetches queueInfo and figures out about the status of the queue, without this the component ends up rendering AdminPageView. I wanted to create a loading state, an error state and a normal state. Hence this code-block.

Reply:
I'm thinking lets continue with normal flow unless we explicitly know that the queue had been deleted.

As discussed, here the only change would be to show the user message and simultaneously take the user to the home page.

Reply:
Sure. Removed loading state.